ruma-events/benches/event_deserialize.rs

```diff
@@ -106,4 +119,10 @@ criterion_group!(
     deserialize_specific_event
 );
 
+#[cfg(feature = "criterion_bench")]
 criterion_main!(benches);
+
+// This makes the compiler only spit out the `compile_error!` when
+// `cargo bench` is run without the `criterion_bench` feature.
+#[cfg(not(feature = "criterion_bench"))]
+fn main() {}
```

Comment:
To have one `#[cfg]` less, we could move the `compile_error!` into the alternative `main`. We could also move all of the actual criterion stuff into a separate `#[cfg]`ed module, though I'm not sure that's worth it (this would require moving `benches/event_deserialize.rs` to `benches/event_deserialize/main.rs`)